gestalt/__init__.py

```diff
@@ -395,15 +395,18 @@ def __get(
             )
         split_keys = key.split(self.__delim_char)
         consider_keys = list()
+        config_val = None
         for split_key in split_keys:
             consider_keys.append(split_key)
             joined_key = self.__delim_char.join(consider_keys)
             config_val = self._get_config_for_key(key=key,
                                                   key_to_search=joined_key,
                                                   default=default,
                                                   object_type=t)
-            if config_val is not None:
+            if config_val is not None and config_val is not default:
```

Comment:
equality checks should be done via `==` instead of `is` here since the `default` value can be a non singleton.